src/mlpack/methods/ann/layer/linear.hpp

```diff
@@ -143,14 +174,21 @@ class LinearType : public Layer<MatType>
   template<typename Archive>
   void serialize(Archive& ar, const uint32_t /* version */);
 
+  // Accessors for inSize and outSize.
+  size_t InputSize() const { return inSize; }
+  size_t OutputSize() const { return outSize; }
```

Comment:
Probably not necessary to expose these---they are only used internally in the forward/backward passes.